jablib/src/main/java/org/jabref/logic/quality/consistency/BibliographyConsistencyCheck.java

```diff
@@ -62,64 +82,115 @@ public record EntryTypeResult(Collection<Field> fields, SequencedCollection<BibE
+        BibDatabaseMode mode = bibContext.getMode();
+        List<BibEntryType> entryTypeDefinitions = (mode == BibDatabaseMode.BIBLATEX)
+                ? BiblatexEntryTypeDefinitions.ALL
+                : BibtexEntryTypeDefinitions.ALL;
 
         int counter = 0;
         for (Map.Entry<EntryType, Set<Field>> mapEntry : entryTypeToFieldsInAnyEntryMap.entrySet()) {
             entriesGroupingProgress.accept(counter++, entryTypeToFieldsInAnyEntryMap.size());
             EntryType entryType = mapEntry.getKey();
-            Set<Field> fields = mapEntry.getValue();
-            Set<Field> commonFields = entryTypeToFieldsInAllEntriesMap.get(entryType);
-            assert commonFields != null;
-            Set<Field> uniqueFields = new HashSet<>(fields);
-            uniqueFields.removeAll(commonFields);
-
-            if (uniqueFields.isEmpty()) {
+            Set<Field> fieldsInAnyEntry = mapEntry.getValue();
+            Set<Field> fieldsInAllEntries = entryTypeToFieldsInAllEntriesMap.get(entryType);
+            assert fieldsInAllEntries != null;
+            Set<Field> differingFields = new LinkedHashSet<>(fieldsInAnyEntry);
+            differingFields.removeAll(fieldsInAllEntries);
+
+            BibEntryType typeDef = entryTypeDefinitions.stream()
+                    .filter(def -> def.getType().equals(entryType))
+                    .findFirst().orElse(null);
+            if (typeDef == null) {
                 continue;
             }
 
-            List<Comparator<BibEntry>> comparators = List.of(
-                    new BibEntryByCitationKeyComparator(),
-                    new BibEntryByFieldsComparator());
-            FieldComparatorStack<BibEntry> comparatorStack = new FieldComparatorStack<>(comparators);
+            Set<Field> requiredFields = typeDef.getRequiredFields().stream()
+                    .flatMap(orFields -> orFields.getFields().stream())
+                    .collect(Collectors.toCollection(LinkedHashSet::new));
 
-            List<BibEntry> differingEntries = entryTypeToEntriesMap
-                    .get(entryType).stream()
-                    .filter(entry -> !filterExcludedFields(entry.getFields()).equals(commonFields))
-                    .sorted(comparatorStack)
-                    .toList();
+            if (mode == BibDatabaseMode.BIBLATEX) {
+                for (Field req : requiredFields) {
+                    if (fieldsInAnyEntry.contains(req) && !fieldsInAllEntries.contains(req)) {
+                        differingFields.add(req);
+                    }
+                }
+            }
 
-            resultMap.put(entryType, new EntryTypeResult(uniqueFields, differingEntries));
+            Set<BibEntry> entries = entryTypeToEntriesMap.get(entryType);
+            if (entries == null || entries.size() <= 1 || differingFields.isEmpty()) {
+                continue;
+            }
+
+            List<BibEntry> sortedEntries;
+            if (mode == BibDatabaseMode.BIBLATEX) {
```

Comment:
Why this? BibTeX also has required and optional fields. -- I think, the if is not required.